src/js/actions/export.js

```diff
@@ -119,6 +120,40 @@ define(function (require, exports) {
     };
 
     /**
+     * Get current status of generator
+     * @private
+     * @return {Promise.<boolean>}
+     */
+    var _getGeneratorStatus = function () {
+        return descriptor.playObject(generatorLib.getGeneratorStatus())
+            .then(function (status) {
+                return objUtil.getPath(status, "generatorStatus.generatorStatus") === 1;
+            });
+    };
+
+    /**
+     * Enable/Disable generator based on supplied parameter
+     * @private
+     * @param {boolean} enabled
+     * @return {Promise}
+     */
+    var _setGeneratorStatus = function (enabled) {
+        return descriptor.playObject(generatorLib.setGeneratorStatus(enabled))
+            .catch(function (e) {
+                throw new Error("Could not enable generator", e);
```

Comment:
Unfortunately I believe the second argument will be lost. I think you either have to stringify it and append it to the message, or set `.cause = e`;